app/src/main/java/com/demoapp/database/DataRepository.kt

```diff
@@ -0,0 +1,89 @@
+package com.demoapp.database
+
+import android.util.Log
+import com.demoapp.model.CategoryEntity
+import com.demoapp.model.DataEntity
+import com.demoapp.model.DataResponse
+import com.demoapp.retrofit.RetrofitInterface
+import retrofit2.Call
+import retrofit2.Callback
+
+class DataRepository(var dataDuo: DataDuo, val mService: RetrofitInterface) {
+
+    fun getLocalData(result: (List<DataEntity>) -> Unit) {
```

Comment:
use suspend function instead of callback

Reply:
added suspend function and removed callback